process_darwin.h

```diff
@@ -46,10 +51,85 @@ static inline int darwinProcesses() {
     resultCount = length / sizeof(struct kinfo_proc);
     for (i = 0; i < resultCount; i++) {
         struct kinfo_proc *single = &result[i];
+        pid_t pid = single->kp_proc.p_pid;
+        char *p, *proc_argv, **argv;
+        size_t count;
+        int nargs, argmax, offset = 0;
+
+        args[1] = KERN_ARGMAX;
+
+        count = sizeof argmax;
+        err = sysctl(args, 2, &argmax, &count, NULL, 0);
+        if (err) {
+            if (errno == EINVAL) {
+                err = 0; errno = 0; nargs = 0;
+                proc_argv = NULL; argv = NULL;
+                goto RESULT;
+            }
+
+            goto ERREXIT;
+        }
+
+        proc_argv = calloc(1, argmax);
+        if (!proc_argv) {
+            goto ERREXIT;
+        }
+
+        args[1] = KERN_PROCARGS2;
+        args[2] = pid;
+
+        count = argmax;
+        err = sysctl(args, 3, proc_argv, &count, NULL, 0);
+        if (err) {
+            // We explicitly return if no valid arguments are found
+            // for this command.
+            if (errno == EINVAL) {
+                err = 0; errno = 0; nargs = 0; argv = NULL;
+                goto RESULT;
+            }
+
+            goto ERREXIT;
+        }
+
+        memcpy(&nargs, proc_argv, sizeof nargs);
+        p = proc_argv + sizeof nargs;
+
+        while (*p != 0) p++;
```

Comment:
This only contains the executable path. The command name will be contained in `argv[0]` as per the standard convention. I will add a comment on this.